### PR TITLE
CI: remove unused imports

### PR DESCRIPTION
**Pull request by de-sh**
Fixes issues found in static analysis with DeepSource

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

Unnecessary imports found on [static analysis of the codebase with DeepSource](https://deepsource.io/gh/de-sh/hydra-python-core/). I used [this DeepSource configuration](https://github.com/de-sh/hydra-python-core/blob/master/.deepsource.toml) to run the static analyzer on the repo, do consider using it if you are to setup the same for CI purposes in the future! 

#### Removed
- Imports made unnecessarily

### CODE REVIEW

File: hydra_python_core/doc_maker.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
 """Contsructor to take a Python dict containing an API Documentation and
 create a HydraDoc object for it
 """
-import re
-import json
 from pyld import jsonld
-import requests
```
Comment (priyanshunayan) [inline hydra_python_core/doc_maker.py]: This import is internally required by the pyld. According to it's docs: 

> Defining a dependency on pyld will not pull in Requests or aiohttp. If you need one of these for a Document Loader then either depend on the desired external library directly or define the requirement as PyLD[requests] or PyLD[aiohttp].

So, we might want to modify either requirements.txt or leave it here.

File: tests/test_doc_maker.py [hunk 2]
```diff
@@ -1,9 +1,6 @@
 import unittest
-import re
-from pyld import jsonld
-import requests
```
Comment (priyanshunayan) [inline tests/test_doc_maker.py:4]: Same as above

**Review by priyanshunayan — CHANGES_REQUESTED** [review]
Just some minor changes. Thank you!

File: hydra_python_core/doc_maker.py (hunk 1 above)
Comment (de-sh) [reply hydra_python_core/doc_maker.py]: Ok, in that case this was a false positive! Thanks for pointing this out :D

**Review by priyanshunayan — APPROVED** [review]
LGTM 👍

**Pull request merged**